lib/client.js

```diff
@@ -88,17 +86,15 @@ class Client extends ClientBase {
       return new PassThrough().destroy(new InvalidArgumentError('invalid handler'))
     }
 
-    let req = new Readable({
+    // TODO: Let ret write directly to socket without intermediate.
+    const req = new PassThrough({
```

Comment:
This should really be a Readable. Otherwise it’s costing us dearly.

Reply:
Ok. I'll give it another try.